azurerm/resource_arm_operational_insight_workspace.go

```diff
@@ -0,0 +1,224 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/operationalinsights"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmOperationalInsightWorkspaceService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmOperationalInsightWorkspaceCreateUpdate,
+		Read:   resourceArmOperationalInsightWorkspaceRead,
+		Update: resourceArmOperationalInsightWorkspaceCreateUpdate,
+		Delete: resourceArmOperationalInsightWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRmOperationalInsightWorkspaceName,
+			},
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"workspace_id": { // a.k.a. customer_id
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"portal_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sku": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"retention_in_days": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"primary_shared_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"secondary_shared_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmOperationalInsightWorkspaceCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).workspacesClient
+	log.Printf("[INFO] preparing arguments for AzureRM Operational Insight workspace creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	skuName := d.Get("sku")
+	sku, err := getSku(skuName)
```

Comment:
since we can just cast directly to the Enum, we can replace this with: 
```
sku := &operationalinsights.Sku{
 Name: operationalinsights.SkuNameEnum(skuName),
}
```

Reply:
Done. Also I removed the getSku and related function.